repobee_sanitizer/_sanitize_repo.py

```diff
@@ -34,11 +34,14 @@ def _sanitize_repo(
         if not args.file_list.is_file():
             raise plug.PlugError(f"No such file: {args.file_list}")
 
-        message = _check_repo_state(args.repo_root)
-        if message:
-            return plug.Result(
-                name="sanitize-repo", msg=message, status=plug.Status.ERROR,
-            )
+        if not args.force:
+            message = _check_repo_state(args.repo_root)
+            if message:
+                return plug.Result(
+                    name="sanitize-repo",
+                    msg=message,
+                    status=plug.Status.ERROR,
+                )
```

Comment:
I prefer not nesting `if` statements.
```suggestion
        message = _check_repo_state(args.repo_root)
        if message and not args.force:
            return plug.Result(
                name="sanitize-repo", msg=message, status=plug.Status.ERROR
            )
```

While one could argue that your implementation is more efficient, it's a most likely imperceptible difference, and the intended use case is to not force anyway, so the common case will be unaltered.

As a fun fact, if we could use Python 3.8, we could write this with the new walrus operator:

```python
        if not args.force and (message := _check_repo_state(args.repo_root)):
            return plug.Result(
                name="sanitize-repo", msg=message, status=plug.Status.ERROR
            )
```
Just as efficient as your solution because of short-circuiting on the `args.force`, and at the same time avoids a nested if. Unfortunately, we maintain 3.6 compatibility, so it's a no-go at the moment.